### DISCUSSION TITLE
RAG + PandasAI methods request

### DISCUSSION THREAD

**Discussion by Tu-Zhenzhao** [sinaptik-ai/pandas-ai#905] [Q&A]
I have prepare the RAG system that ready to do any vector search and retrieve, but I want work it with PandasAI to make the prompt more accrete and help PandasAI to generate better code result. Is any method like pipeline in Langchain that allow us to connect personal RAG with PandasAI?

**Comment by ssling0817** [comment]
@Tu-Zhenzhao Do you get any solution or workaround for this? I also want to implement RAG with PandasAI, thank you!

**Comment by gventuri** [comment]
Check out: https://docs.pandas-ai.com/train

However PandasAI doesn't properly support RAG, it only uses RAG for in-context learning. So it's not meant to handle large documents etc, but rather specific instructions or question-answer pairs.

**Reply by ssling0817** [reply]
Thanks for your response. Is there any workaround to use PandasAI with RAG?

**Reply by gventuri** [reply]
Not at the moment, as PandasAI is not currently meant to be a retrieval library. But if you can share more about your use case, we'd love to know more so that we can add it to the roadmap :)
